martin-mbtiles/src/tile_copier.rs

```diff
@@ -186,8 +191,40 @@ impl TileCopier {
     }
 
     async fn copy_tile_tables(&self, conn: &mut SqliteConnection) -> MbtResult<()> {
-        self.run_query_with_options(conn, "INSERT INTO tiles SELECT * FROM sourceDb.tiles")
+        if let Some(diff_file) = &self.options.diff_with_file {
+            let diff_mbtiles = Mbtiles::new(diff_file)?;
+
+            let _ = &self.detect_type(&diff_mbtiles);
```

Comment:
this is a bug:  `detect_type` returns an awaitable, which you must await in order to let it detect the type. But instead you ignore the result, thus it will never even execute.  I have renamed detect_type in a new patch, but left it there because I am not certain why you have it - in theory the diff file shouldn't even exist yet, should it?

Reply:
This file we are diffing against does exist and this line was meant to ensure that it was a vaild `mbtiles` type. There is a bug though, I should have added an await; thank you for updating!